src/hotspot/share/jfr/periodic/sampling/jfrCPUTimeThreadSampler.cpp

```diff
@@ -163,20 +163,40 @@ void JfrCPUTimeTraceQueue::clear() {
   Atomic::release_store(&_head, (u4)0);
 }
 
-static int64_t compute_sampling_period(double rate) {
-  if (rate == 0) {
-    return 0;
+// A throttle is either a rate or a fixed period
+class JfrCPUSamplerThrottle {
+
+  double _rate;
+  u8 _period_nanos;
+  bool _is_rate;
```

Comment:
Can this be turned into `union` - only one of `_rate`  or `_period_nanos` will ever be set at a time.